JSIL/AST/JSAstVisitor.cs

```diff
@@ -19,6 +19,21 @@ public struct EnclosingNode<T> {
             public string ChildName;
         }
 
+        public class NodeProcessedEventArg : EventArgs
```

Comment:
Can you make this a struct or just pass a JSNode directly? 'sender' is useless here so there's not much point in paying for a heap allocation per-node.